mock-plugin/src/test/java/io/mosip/signup/plugin/mock/service/MockProfileRegistryPluginImplTest.java

```diff
@@ -0,0 +1,272 @@
+        verifiedData.put("UIN","1234567890");
+        verifiedData.put("individualId",individualId);
+
+        JsonNode mockIdentity = objectMapper.valueToTree(verifiedData);
+        ResponseWrapper<JsonNode> responseWrapper = new ResponseWrapper<>();
+        responseWrapper.setResponse(mockIdentity);
+        ResponseEntity<ResponseWrapper<JsonNode>> responseEntity=new ResponseEntity<>(responseWrapper, HttpStatus.OK);
+
+        Mockito.when(restTemplate.exchange(
+                "http://localhost:8080/"+individualId,
+                HttpMethod.GET,
+                null,
+                new ParameterizedTypeReference<ResponseWrapper<JsonNode>>() {
+                })).thenReturn(responseEntity);
+        ProfileDto profileDto= mockProfileRegistryPlugin.getProfile(individualId);
+        Assert.assertNotNull(profileDto);
+        Assert.assertEquals(profileDto.getIndividualId(),"1234567890");
+    }
+
+    @Test
+    public void getProfile_withErrorCodeAsIndividualIdFail_thenFail() throws ProfileException {
+        String individualId = "1234567890";
+        ReflectionTestUtils.setField(mockProfileRegistryPlugin, "getIdentityEndpoint","http://localhost:8080/");
+
+        Map<String, Object> verifiedData = new HashMap<>();
+        verifiedData.put("email","123@email.com");
+        verifiedData.put("password","123456");
+        verifiedData.put("UIN","1234567890");
+        verifiedData.put("individualId",individualId);
+
+        JsonNode mockIdentity = objectMapper.valueToTree(verifiedData);
+        ResponseWrapper<JsonNode> responseWrapper = new ResponseWrapper<>();
+        responseWrapper.setResponse(mockIdentity);
+        Error error = new Error();
+        error.setErrorCode("invalid_individual_id");
+        responseWrapper.setErrors(List.of(error));
+        ResponseEntity<ResponseWrapper<JsonNode>> responseEntity=new ResponseEntity<>(responseWrapper, HttpStatus.OK);
+
+        Mockito.when(restTemplate.exchange(
+                "http://localhost:8080/"+individualId,
+                HttpMethod.GET,
+                null,
+                new ParameterizedTypeReference<ResponseWrapper<JsonNode>>() {
+                })).thenReturn(responseEntity);
+
+        ProfileDto profileDto= mockProfileRegistryPlugin.getProfile(individualId);
+        Assert.assertNotNull(profileDto);
+        Assert.assertEquals(profileDto.getIndividualId(),individualId);
+    }
+
+
+    @Test
+    public void updateProfile_withVerifiedClaim_thenPass()  {
+
+        ReflectionTestUtils.setField(mockProfileRegistryPlugin, "objectMapper",objectMapper);
+        ReflectionTestUtils.setField(mockProfileRegistryPlugin, "addVerifiedClaimsEndpoint","http://localhost:8080/");
+        String requestId = "req-123";
+        String individualId = "ind-456";
+
+        Map<String, Object> verifiedData = new HashMap<>();
+        verifiedData.put("email","123@email.com");
+        verifiedData.put("password","123456");
```

Comment:
Although its verified data, example data doesnot illustrate what is expected to be verifiedData. It should have verification details like trustFramework, processName and so on.

Reply:
added some extra details